org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/FileTextEditConverter.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.jdt.ls.extension.core.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.text.edits.CopyTargetEdit;
+import org.eclipse.text.edits.DeleteEdit;
+import org.eclipse.text.edits.InsertEdit;
+import org.eclipse.text.edits.MalformedTreeException;
+import org.eclipse.text.edits.MoveSourceEdit;
+import org.eclipse.text.edits.MoveTargetEdit;
+import org.eclipse.text.edits.ReplaceEdit;
+import org.eclipse.text.edits.TextEdit;
+import org.eclipse.text.edits.TextEditVisitor;
+
+/**
+ * Converts an {@link org.eclipse.text.edits.TextEdit} to {@link org.eclipse.lsp4j.TextEdit} for
+ * {@link IDocument}.
+ *
+ * @author Valeriy Svydenko
+ */
+public class FileTextEditConverter extends TextEditVisitor {
+
+  private TextEdit source;
+  private IDocument document;
+  private List<org.eclipse.lsp4j.TextEdit> converted;
+
+  public FileTextEditConverter(IDocument document, TextEdit edit) {
+    this.source = edit;
+    this.converted = new ArrayList<>();
+    if (document == null) {
+      throw new IllegalArgumentException("Document can not be null");
+    }
+    this.document = document;
+  }
+
+  public List<org.eclipse.lsp4j.TextEdit> convert() {
```

Comment:
What if someone calls this method twice? I ususally solve the problem by making a public static method 
```
public static List<TextEdit> convert(document, edit) 
```
which creates an instance of FileTextEditConverter and calls convert on that. Then the constructor and the non-static convert method can be private. This way, you can't misuse the API.

Reply:
fixed, thx!